src/app/modules/item/task/task-xd-pr.ts

```diff
@@ -0,0 +1,303 @@
+  }
+
+  reloadState(state : string) : Observable<void> {
+    return this.chan.call({
+      method: "task.reloadState",
+      params: state,
+      timeout: 2000
+    });
+  }
+
+  getViews() : Observable<TaskViews> {
+    // TODO: validator
+    return this.chan.call({
+      method: "task.getViews",
+      timeout: 2000
+    });
+  }
+
+  showViews(views : Object) : Observable<void> {
+    return this.chan.call({
+      method: "task.showViews",
+      params: views,
+      timeout: 2000
+    });
+  }
+
+  gradeAnswer(answer : string, answerToken : string) : Observable<TaskGrade> {
+    // TODO: validator
+    return this.chan.call({
+      method: "task.gradeAnswer",
+      params: [ answer, answerToken ],
+      timeout: 40000
+    });
+  }
+
+  getResources() : Observable<TaskResources> {
+    // TODO: validator
+    return this.chan.call({
+      method: "task.getResources",
+      params: [],
+      timeout: 2000
+    });
+  }
+}
+
+/*
+ * Platform object definition, created from a Task object (see below)
+ */
+
+export class Platform {
+  task : Task;
+  constructor(task : Task) {
+    this.task = task;
+  }
+
+  getTask() : Task {
+    return this.task;
+  }
+
+  /*
+   * Simple prototypes for platform API functions, to be overriden by your
+   * platform's specific functions (for each platform object)
+   */
+
+  validate(_mode : string) : Observable<void> {
+    // TODO: validator
+    return throwError(() => new Error('platform.validate is not defined'));
+  }
+  showView(_views : any) : Observable<void> {
+    // TODO: validator
+    return throwError(() => new Error('platform.showView is not defined'));
+  }
+  askHint(_platformToken : string) : Observable<void> {
+    // TODO: validator
+    return throwError(() => new Error('platform.validate is not defined'));
+  }
+  updateHeight(height : number) : Observable<void> {
+    // TODO: validator
+    return this.updateDisplay({ height: height });
+  }
+  updateDisplay(_data : UpdateDisplayParams) : Observable<void> {
+    // TODO: validator
+    return throwError(() => new Error('platform.updateDisplay is not defined!'));
+  }
+  openUrl(_url : string) : Observable<void> {
+    // TODO: validator
+    return throwError(() => new Error('platform.openUrl is not defined!'));
+  }
+  log(_data : TaskLog) : Observable<void> {
+    // TODO: validator
+    return throwError(() => new Error('platform.log is not defined!'));
+  }
+  getTaskParams(keyDefault? : [string, TaskParamsValue]) : Observable<TaskParamsValue> {
+    // TODO: validator
+    const key = keyDefault ? keyDefault[0] : undefined;
+    const defaultValue = keyDefault ? keyDefault[1] : undefined;
+    const res : {[key: string]: TaskParamsValue} = { minScore: -3, maxScore: 10, randomSeed: 0, noScore: 0, readOnly: false, options: {} };
+    if (key) {
+      if (key !== 'options' && key in res) {
+        return of(res[key]);
+      } else {
+        return of(defaultValue);
+      }
+    }
+    return of(res);
+  }
+}
```

Comment:
I don't see the reason to make them observable. From what I understand, there are signal from the task. So they can be treated with no delay. 
`getTaskParams` shows that, observables look useless there.

Reply:
They can't always be treated with no delay, for instance when it comes to validate or askHint which require calls to the backend and/or the task. Moreover, for instance for validate, making it an observable allows to keep the rxjs logic, [example here with this first version of validate](https://github.com/France-ioi/AlgoreaFrontend/blob/1478336586c442642df72db451b669d4978ea08d/src/app/modules/item/pages/item-display/item-display.component.ts#L164).

As some seem better to be written with Observables, I just made everything Observable to avoid having to write different definitions for `channel.bind`.

Reply:
Okay for the delay. 
For `getTaskParams` which returns a response, I think `bind` does not do anything with the observable result, does it?

Reply:
`bind` [sends the result to the task through jschannel](https://github.com/France-ioi/AlgoreaFrontend/pull/525/files#diff-b0866128bd3efd33f554be8000bf7103073584b8206ad42a2a1a1297960b77caR48).